### ISSUE TITLE
Custom Weighted Absolute Error loss does not work as expected

### ISSUE THREAD

**Issue by borjaf696** [microsoft/LightGBM#6188]
## Description

I have defined a weighted version of the L1 loss function but it is not working as expected:
```python
def custom_mae(y_true, y_pred, weight):
    gradient = weight * np.sign(y_pred - y_true)
    hessian = np.zeros_like(y_true)
    return gradient, hessian  
```

Just to try if I was doing something wrong I tried the weighted L2 and it worked perfectly:
```python
def custom_l2(y_true, y_pred, weight):
    grad = (y_pred - y_true)*weight
    hess = np.ones(len(y_true))
    return grad, hess
```

* Another problem that I have is that the `fit` creates a lot of logs with:
`[LigthGBM] [Warning] No Further splits with ...`
This bug has been reported in the past but the solution of using `verbose = -1` in the params for the model is not working for me
## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell

```

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

Could you please be more specific about what "not working as expected" means, and provide a minimal, reproducible example ([docs on what that is](https://stackoverflow.com/help/minimal-reproducible-example)) showing how your determined that?

If you're unsure how to create a minimal, reproducible example of using the Python package, look around this project's issues for posts using `sklearn.datasets`, e.g. https://github.com/microsoft/LightGBM/issues/4239#issuecomment-883928397.

Please also provide the information that you were asked to provide when you clicked `New Issue`:

* what version of `lightgbm`?
* how did you install it?
* what operating system?
